src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -335,6 +349,33 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm)
                 if (string.IsNullOrEmpty(searchTerm))
                     continue;
 
+                bool firstTermIsCategory = false;
+
+                //This will only valid when the search term has empty spaces
+                if (hasEmptySpaces)
+                {
+                    //Check if the first term of the search criteria match any category
+                    var possibleCategory = searchTerm.Split(' ')[0];
+                    firstTermIsCategory = dynamoModel.SearchModel.Entries.Where(entry => IsMatchingCategory(possibleCategory, entry.FullCategoryName) == true && !string.IsNullOrEmpty(possibleCategory)).Any();
+
+                    //Get the node matching the Category provided in the search term
+                    var matchingCategory = dynamoModel.SearchModel.Entries.Where(entry => IsMatchingCategory(possibleCategory, entry.FullCategoryName) == true && !string.IsNullOrEmpty(possibleCategory)).FirstOrDefault();
```

Comment:
The same Where function on Entries is running twice, store the result and use it again.

Reply:
Also can we check if the possibleCategory is null or empty before the where call.

Reply:
Fixed in the next commit:
https://github.com/DynamoDS/Dynamo/pull/14917/commits/c62ba640baa2c7623d3711f74f27c6ad9f227c47